### ISSUE TITLE
Carmine v4 Cluster support

### ISSUE THREAD

**Issue by ptaoussanis** [taoensso/carmine#298]

**Comment by charan1973** [comment]
Hi @ptaoussanis. Can I pick this up? I can reach out to you on Slack if you need to discuss anything.

**Comment by ptaoussanis** [comment]
@charan1973 Hi Charan, thanks for pinging about this!

What's your level of familiarity with Redis Cluster and the Redis protocol? I've got some of the groundwork already laid for Cluster support, but have a few non-trivial parts that still need to be implemented.

And don't currently use (or have any use for) Cluster myself, so testers would definitely be helpful + welcome later.

**Comment by charan1973** [comment]
At my current job, we use redis cluster for managing data. So I do have familiarity to work with it. I can setup and test on my machine.

Please let me know how I can help or any other details if needed.

**Comment by charan1973** [comment]
Thanks for the reply. The company service has been already moved to Jedis(I am not representing them). I liked carmine as it was easier to read compared to Jedis so thought of contributing during my free time and learn more. 

No worries if you are working on it. Thanks 🙏🏼

**Comment by ptaoussanis** [comment]
Ah, gotcha 👍

So I'm definitely not actively working on it right now. Carmine v4's a huge amount of work, and there's been little interest in it expressed by the community - so I've mostly been working on it as a background task.

Most of the [core](https://github.com/taoensso/carmine/blob/master/src/taoensso/carmine_v4.clj) is already usable, as is [Sentinel](https://github.com/taoensso/carmine/blob/master/src/taoensso/carmine_v4/sentinel.clj) support. I've recently deployed a project to production using the new core.

But there'd still be a ton of work (esp. documentation) to make this useful to typical users.

The [Cluster implementation](https://github.com/taoensso/carmine/blob/master/src/taoensso/carmine_v4/cluster.clj) is the least developed since I have the least use for it myself. The core has been designed with Cluster support [in mind](https://github.com/taoensso/carmine/blob/b0a8df08dccc49feff9184032786b2cbef78eed9/src/taoensso/carmine_v4/resp.clj#L318) though.

Basically: getting v4 usable, especially with Cluster support - would still be a lot of additional design and implementation work. Both would be non-trivially complex, so _probably_ not an ideal candidate for user contributions unless the contributors are prepared to dedicate a lot of time+effort.

It's helpful though to see folks express interest and to offer to help test. I'm still planning my open source roadmap for the year, and definitely would like to get v4 publicly usable at some point. Just a matter of timing/prioritisation - have several big-ticket items I'd like to get done this year.
